### ISSUE TITLE
Importing k2 on MacOS fails: image not found

### ISSUE THREAD

**Issue by pzelasko** [k2-fsa/k2#836]
I installed k2 with the following method:

```
conda install -c k2-fsa -c pytorch cpuonly k2 python=3.6 pytorch=1.8.1
```

When running Icefall yesno recipe to test it out, I got the following error:

```
 ./prepare.sh
2021-09-29 15:26:46 (prepare.sh:24:main) dl_dir: /Users/pzelasko/jhu/icefall/egs/yesno/ASR/download
2021-09-29 15:26:46 (prepare.sh:27:main) stage 0: Download data
2021-09-29 15:26:46 (prepare.sh:36:main) Stage 1: Prepare yesno manifest
2021-09-29 15:26:47 (prepare.sh:42:main) Stage 2: Compute fbank for yesno
Traceback (most recent call last):
  File "./local/compute_fbank_yesno.py", line 18, in <module>
    from icefall.utils import get_executor
  File "/Users/pzelasko/jhu/icefall/icefall/utils.py", line 28, in <module>
    import k2
  File "/Users/pzelasko/miniconda3/envs/icefall/lib/python3.6/site-packages/k2/__init__.py", line 2, in <module>
    from _k2 import DeterminizeWeightPushingType
ImportError: dlopen(/Users/pzelasko/miniconda3/envs/icefall/lib/python3.6/site-packages/_k2.cpython-36m-darwin.so, 2): Library not loaded: @rpath/libk2context.dylib
  Referenced from: /Users/pzelasko/miniconda3/envs/icefall/lib/python3.6/site-packages/_k2.cpython-36m-darwin.so
  Reason: image not found
```

BTW I also tried with python=3.8, but conda stated there are version conflicts (i.e., the binaries for python 3.8 don't seem to be uploaded) -- is this expected?

**Comment by csukuangfj** [comment]
>   Reason: image not found

I also encountered this error before and have noted down the solution in the documentation.
Please see https://k2-fsa.github.io/k2/installation/conda.html

The short answer is
```
export DYLD_LIBRARY_PATH=$CONDA_PREFIX/lib/python3.8/site-packages:$DYLD_LIBRARY_PATH
```

I don't know where the error comes from.


---

> BTW I also tried with python=3.8, but conda stated there are version conflicts (i.e., the binaries for python 3.8 don't seem to be uploaded) -- is this expected?

GitHub actions start to fail for Python 3.7 and 3.8 at some point. 
Please see https://github.com/k2-fsa/k2/actions/runs/1249759906

I will have a look at it today.

If you want to use python3.8 on macOS, please use `pip install`.
See https://k2-fsa.github.io/k2/installation/pip.html
and
https://k2-fsa.org/nightly/index.html

**Comment by pzelasko** [comment]
Thanks for help! 

BTW maybe it makes sense to try and catch this error and display the solution for the users? Like this:

In k2's `__init__.py`:

https://github.com/k2-fsa/k2/blob/601d663fa11dbab0788f41013aee08db62c7ac18/k2/python/k2/__init__.py#L2-L3

wrap some (all?) imports with `try/except`:

```python
try:
    from _k2 import DeterminizeWeightPushingType 
    from _k2 import simple_ragged_index_select 
except ImportError as e:
    raise ImportError(str(e) + "\nIf you're importing k2 on MacOS, "
        "you might want to set the environment variable:\nexport DYLD_LIBRRARY_PATH=...")
```

**Comment by csukuangfj** [comment]
> BTW maybe it makes sense to try and catch this error and display the solution for the users? Like this:

Yes, I agree. It provides the solution directly to the user.

**Comment by csukuangfj** [comment]
> the binaries for python 3.8 don't seem to be uploaded

I suspect that the reason for the failure of the GitHub acitons is that mkl is updated from 2021.2.0 to 2021.3.0
It was OK for mkl 2021.2.0. I am trying to fix the version of mkl to 2021.2.0 to see whether it solves the problem.
